apps/api/src/domain/repositories/interfaces/IEdgeConnectionRepository.ts

```diff
@@ -0,0 +1,22 @@
+import { Maybe } from '../../../lib/types/maybe';
+import { ResultOrError } from '../../../types/ResultOrError';
+import { EdgeConnection } from '../../models/context/edge-connection.entity';
+import { EntityId } from '../../types/ResourceId';
+import { ISpecification } from './ISpecification';
+
+export interface IEdgeConnectionRepository {
+    /**
+     * Find one edge connection by ID.
+     *
+     * @returns EdgeConnection or symbol(NotFound)
+     */
+    fetchById: (id: EntityId) => Promise<ResultOrError<Maybe<EdgeConnection>>>;
+
+    // Returns an empty array if none found
+    fetchMany: (
+        specification?: ISpecification<EdgeConnection>
+    ) => Promise<ResultOrError<EdgeConnection>[]>;
+
+    // Do we want getStats instead?
+    getCount: () => Promise<number>;
```

Comment:
Do we want the 'write' methods here as well? I guess we can opt into them when we need them, but be sure to program to the interface not the concrete implementation in the domain \ view model layers.